rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -0,0 +1,359 @@
+
+		// Compare current config value with the one in this user (only 8 digits in userId)
+		if (currentUserId.compare(0, 8, user->GetUserId()) == 0)
+		{
+			userIdItem->setFont(boldFont);
+			userNameItem->setFont(boldFont);
+		}
+		++row;
+	}
+
+	// GUI resizing
+	m_table->horizontalHeader()->resizeSections(QHeaderView::ResizeToContents);
+	m_table->verticalHeader()->resizeSections(QHeaderView::ResizeToContents);
+
+	QSize tableSize = QSize(
+		m_table->verticalHeader()->width() + m_table->horizontalHeader()->length() + m_table->frameWidth() * 2,
+		m_table->horizontalHeader()->height() + m_table->verticalHeader()->length() + m_table->frameWidth() * 2);
+
+	QSize preferredSize = minimumSize().expandedTo(sizeHint() - m_table->sizeHint() + tableSize).expandedTo(size());
+	QSize maxSize = QSize(preferredSize.width(), static_cast<int>(QApplication::desktop()->screenGeometry().height()*.6));
+
+	resize(preferredSize.boundedTo(maxSize));
+}
+
+//Remove a user folder, needs to be confirmed.
+void user_manager_dialog::OnUserRemove()
+{
+	int idx = m_table->currentRow();
+	int idx_real = m_table->item(idx, 0)->data(Qt::UserRole).toInt();
+	if (QMessageBox::question(this, tr("Delete Confirmation"), tr("Are you sure you want to delete:\n%1?").arg(qstr(m_user_list[idx_real]->GetUserName())), QMessageBox::Yes, QMessageBox::No) == QMessageBox::Yes)
+	{
+		fs::remove_all(m_user_list[idx_real]->GetUserDir());
+		UpdateTable();
+	}
+}
+
+void user_manager_dialog::GenerateUser(const std::string& username)
+{
+	// If the user list is sorted by defult from fs::dir, then we just need the last one in the list.
+	std::string largestUserId = m_user_list[m_user_list.size() - 1]->GetUserId();
+
+	// Add one to the largest user id, then reformat the result into an 8-digit string.
+	u8 nextLargest = static_cast<u8>(std::stoul(largestUserId) + 1u);
+	char* buf;
+	sprintf(buf, "%08d", nextLargest);
+	std::string nextUserId(buf);
+
+	// Create user folders and such.
+	const std::string homeDir = Emu.GetHddDir() + "home/";
+	const std::string userDir = homeDir + nextUserId;
+	fs::create_dir(homeDir);
+	fs::create_dir(fmt::format("%s/", userDir));
+	fs::create_dir(fmt::format("%s/exdata/", userDir));
+	fs::create_dir(fmt::format("%s/savedata/", userDir));
+	fs::create_dir(fmt::format("%s/trophy/", userDir));
+	fs::write_file(fmt::format("%s/localusername", userDir), fs::create + fs::excl + fs::write, username);
+}
+
+bool user_manager_dialog::ValidateUsername(const QString& textToValidate)
+{
+	// "Entire string (^...$) must be between 3 and 16 characters
+	// and only consist of letters, numbers, underscores, and hyphens."
+	QRegExpValidator validator(QRegExp("^[A-Za-z0-9_-]{3,16}$"));
+
+	int pos = 0;
+	QString text = textToValidate;
+	return (validator.validate(text, pos) == QValidator::Acceptable);
+}
+
+void user_manager_dialog::OnUserRename()
+{
+	user_create_dialog* renameDialog = new user_create_dialog(tr("Rename User"), this);
```

Comment:
The whole user_create_dialog class is pretty much useless.
You can just use a QInputDialog like pirhoman did in `settings_dialog::OnBackupCurrentConfig`.
The only downside to that would be that we cannot use the `setMaxLength(16)` if I remember correctly.

Reply:
If my regular expression works as intended, that won't be a problem.

Reply:
And yeah, that's what I get for not trying to find the best pre-defined class for this kind of work. Shame on me!